test/functional/ex_cmds/excmd_spec.lua

```diff
@@ -50,4 +52,22 @@ describe('Ex cmds', function()
     eq(2, funcs.exists(':defer'))
     eq('defer', funcs.fullcommand('defer'))
   end)
+
+  it(':.! should pipe line buffer to &shell', function()
+    if not helpers.has_powershell() then
+      insert('find . \'src\' -maxdepth 0 -type d')
+      command(':.!')
+      local res = curbuf_contents()
+      eq(res, '.\nsrc')
+    end
+  end)
+
+  it(':%! should pipe buffer to &shell', function()
+    if not helpers.has_powershell() then
+      insert('find . \'src\' -maxdepth 0 -type d;\necho \'hello world\';')
```

Comment:
can use `"` to avoid needing to escape stuff
```suggestion
      insert([[find . 'src' -maxdepth 0 -type d;\necho 'hello world';]])
```

Reply:
ideally we'd also have a powershell-specific assertion here, in a `else`